watcher/src/bin/main.rs

```diff
@@ -147,9 +149,19 @@ impl WatcherSyncThread {
                 break;
             }
 
+            // Determine from config how we should sync watcher
+            let sync_limit = if let Some(max_block_height) = config.max_block_height {
+                SyncLimit::MaxBlockHeight {
+                    max_block_height,
+                    max_blocks_per_iteration: None,
```

Comment:
I believe this is problematic because the stop trigger will never get checked. Is there a reason to not limit the number of blocks per iteration to 1000 so that every (at most) 1000 blocks we check if the thread has been ordered to stop?

Reply:
If I'm not Mistaken, wouldn't that create the problem cited above with the sleeps causing the sync to take longer than necessary?

> I think this is suboptimal - it will result the code in https://github.com/mobilecoinfoundation/mobilecoin/blob/master/watcher/src/bin/main.rs#L150-L159 taking longer to execute since there will be more sleeps in between each batch.

Reply:
I see, yeah, you're right. I am not sure what the optimal solution is, but one option I can think of is to change `sync_blocks` to `AllUrlsFailed`, `NothingToSync`, `ReachedMaxBlocksPerIteration` etc to cover all the possible return cases. Then the caller can sleep skip sleeping when iteration count is exceeded, but sleep is everything is synced. Note that I have not fully thought this through... But the desired solution here is that:
1) sync_blocks can be called without sleeps as long as there are blocks to sync
2) sync_blocks can support limiting how many blocks it syncs in a single call to allow checking for stop triggers
3) sync_blocks can notify the caller when there's no point in calling it again immediately, in which case the caller can sleep
4) sync_blocks doesn't exhibit the original bug where a single node could halt the syncing of all other ones

Reply:
Yeah I'm on the same page as you on this on points 1-4, it should be simple to achieve that, I'll iterate on this!

Reply:
Thank you!

Reply:
Revised it to use a result enum to handle the cases above and just passed the two params as direct arguments